### ISSUE TITLE
Make PyTorch Metric Learning loss functions registerable

### ISSUE THREAD

**Issue by JohnGiorgi** [JohnGiorgi/t2t#34]
WIP.
